### PR TITLE
Restructuring of build process

### PR DESCRIPTION
**Pull request by tboerger**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,11 @@
-# GOARCH=amd64 GOOS=linux CGO_ENABLED=0 go build -o drone-volume-cache
-# docker build --rm -t plugins/volume-cache .
-FROM scratch
+FROM drone/base:scratch
+MAINTAINER Drone.IO Community <brad@drone.io>
```
Comment (bradrydzewski) [inline Dockerfile]: this plugin can use `FROM scratch` because it doesn't use the network

File: main.go [hunk 2]
```diff
@@ -9,14 +10,14 @@ import (
...
-var version = "1.0.0" // build number set at compile-time
+var build = "0" // build number set at compile-time
 
 func main() {
 	app := cli.NewApp()
 	app.Name = "volume cache plugin"
 	app.Usage = "valume cache plugin"
 	app.Action = run
-	app.Version = version
+	app.Version = fmt.Sprintf("1.0.%s", build)
```
Comment (bradrydzewski) [inline main.go]: maybe `1.0.0+%s` based on the semver examples?

> Thus two versions that differ only in the build metadata, have the same precedence. Examples: 1.0.0-alpha+001, 1.0.0+20130313144700, 1.0.0-beta+exp.sha.5114f85.

Comment (tboerger) [reply main.go]: Sure, I just took this from other plugins

File: .dockerignore [hunk 3]
```diff
@@ -1,8 +1,2 @@
-.git
-.gitignore
-.env
-*.yml
-*.go
-*.md
-*.sh
-LICENSE
+*
+!drone-volume-cache
```
Comment (bradrydzewski) [inline .dockerignore]: much cleaner 👍

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Review by josmo — APPROVED** [review]
LGTM

**Review by tonglil — APPROVED** [review]

**Pull request merged**